core/src/modules/DynamicsModule/BBMDynamics.cpp

```diff
@@ -108,26 +108,28 @@ void BBMDynamics::setData(const ModelState::DataMap& ms)
             kernel.setData(fieldName, mask(data));
         }
     }
+
+    // Set the DG field data
+    kernel.setDGArray(hiceName, hiceDG.allComponents());
+    kernel.setDGArray(ciceName, ciceDG.allComponents());
 }
 
 void BBMDynamics::update(const TimestepTime& tst)
 {
     std::cout << tst.start << std::endl;
 
     // Fill the updated damage array with the initial value
-    damage = damage0.data();
+    damage = damage0;
 
     // set the updated ice thickness, concentration and damage
-    kernel.setData(hiceName, hice.data());
-    kernel.setData(ciceName, cice.data());
     kernel.setData(damageName, damage);
```

Comment:
We need `setData` here (and `getData` below) because m_damage is HField and not AdvectionField in PrognosticData - right?

Then that's a good reason to change the type of m_damage, because using damage should be the default.

Reply:
Perhaps more accurately, the damage is not yet a `DGVectorHolder` in the dynamics. But it amounts to the same thing. Other advected fields are my next target.